src/comments.js

```diff
@@ -195,13 +195,33 @@ function handleForComments(
 }
 
 function handleClassComments(enclosingNode, followingNode, comment) {
-  if (enclosingNode && enclosingNode.kind === "class") {
+  if (
+    enclosingNode &&
+    ["class", "interface", "trait"].includes(enclosingNode.kind)
+  ) {
     // for extends nodes that have leading comments, we can store them as
     // dangling comments so we can handle them in the printer
-    if (followingNode === enclosingNode.extends) {
-      addDanglingComment(followingNode, comment);
-      return true;
+
+    if (followingNode && enclosingNode.extends) {
+      if (!Array.isArray(enclosingNode.extends)) {
+        if (followingNode === enclosingNode.extends) {
+          addDanglingComment(followingNode, comment);
+          return true;
+        }
+      } else {
+        if (
```

Comment:
`else if`

Reply:
It will be necessary here to rewrite something here so I will correct in the following PRs